pkg/apis/core/helper/helpers_test.go

```diff
@@ -653,6 +653,39 @@ var _ = Describe("helper", func() {
 		previewVersion,
 	}
 
+	var machineImages = []core.MachineImage{
+		{
+			Name: "coreos",
+			Versions: []core.MachineImageVersion{
+				{
+					ExpirableVersion: core.ExpirableVersion{
+						Version: "0.0.2",
+					},
+				},
+				{
+					ExpirableVersion: core.ExpirableVersion{
+						Version: "0.0.3",
+					},
+				},
+			},
+		},
+	}
+
+	DescribeTable("#DetermineLatestMachineImageVersions",
+		func(versions []core.MachineImage, expectation map[string]core.MachineImageVersion, expectError bool) {
+			result, err := DetermineLatestMachineImageVersions(versions)
+			if expectError {
+				Expect(err).To(HaveOccurred())
```

Comment:
Wouldn't it be better to remove the if and have something like this:
```go
	Expect(err).To(errorMatcher)
	Expect(result).To(Equal(expectation))
```
Then pass `Succeed()` or `HaveOccurred()` for the `errorMatcher`, and `nil` or the expected map for `expectation` 

This way we correctly perform the error check when we expect that it should not occur, instead of skipping it.

Reply:
There are other test in this file following the same approach so we can keep it for now.